test/load_arbitrary_tool/load_arbitrary_tool_test.bzl

```diff
@@ -19,7 +14,7 @@ def _load_arbitrary_tool_test_impl(repository_ctx):
         tool_subdirectories = ["cargo"],
         version = "1.53.0",
         iso_date = None,
-        target_triple = target_triple,
+        target_triple = host_triple.str,
```

Comment:
I added a `str` field to the struct returned by `triple("...")`, because `triple.triple` didn't strike me as evocative enough. `str` makes sense to me as the string representation of the triple, which is what `target_triple` wants.

Alternatively we could look into making `target_triple` accept the triple struct directly, or union of triple struct and string so that either way works.

Reply:
> I added a str field to the struct returned by triple("..."), because triple.triple didn't strike me as evocative enough. str makes sense to me as the string representation of the triple, which is what target_triple wants.

I'm fine with the use of `str` but I'd prefer there to be either `str` or `triple`. Could you maybe update the docstring to indicate which of the two is the correct to use?

> I added a `str` field to the struct returned by `triple("...")`, because `triple.triple` didn't strike me as evocative enough. `str` makes sense to me as the string representation of the triple, which is what `target_triple` wants.
> 
> Alternatively we could look into making `target_triple` accept the triple struct directly, or union of triple struct and string so that either way works.
> 
> Or we could make `get_host_triple` return the string and not the struct, forcing `triple(get_host_triple(repository_ctx))` when someone needs the struct.

I think I would prefer to continue to pass the `triple` struct until we find a performance reason to do otherwise. We have a lot of places where we parse the same string over and over. I expect passing a struct to be slightly better.